### ISSUE TITLE
No way to make a radio button's label UPPERCASE

### ISSUE THREAD

**Issue by pzahemszky** [enthought/traitsui#463]
When using an Enum trait and displaying it as a radio button with EnumEditor, the automatic formatting makes the first letter of the labels capitalised and the rest lowercase. This is still the case when the `values` argument is used for EnumEditor (see the example code below).
```
from traits.api import Enum, HasTraits
from traitsui.api import EnumEditor, Item, View

class EnumExample(HasTraits):
    priority = Enum('UPPERCASE', 'lowercase')

    view = View(
        Item(
            'priority',
            editor=EnumEditor(
                values={
                    'UPPERCASE': '1:UPPERCASE',
                    'lowercase': '2:lowercase',
                },

            ),
            style='custom',
        )
    )


if __name__ == '__main__':
    EnumExample().configure_traits()
```
Note: the behaviour is the same with the `values` argument being `{'Uppercase': '1:UPPERCASE', 'Lowercase': '2:lowercase'}` (i.e. mapping the displayed default spelling to the desired one).

Maybe related to #267.

**Comment by corranwebster** [comment]
A work-around is the use of a do-nothing `format_func` argument:
```
import six

from traits.api import Enum, HasTraits
from traitsui.api import EnumEditor, Item, View

class EnumExample(HasTraits):
    priority = Enum('UPPERCASE', 'lowercase')

    view = View(
        Item(
            'priority',
            editor=EnumEditor(
                values={
                    'UPPERCASE': '1:UPPERCASE',
                    'lowercase': '2:lowercase',
                },
                format_func=six.text_type,
            ),
            style='custom',
        )
    )


if __name__ == '__main__':
    EnumExample().configure_traits()
```

**Comment by corranwebster** [comment]
And looking at the code I think that, given the work around and the amount of disruption to the code to refactor away from this, we'll call the fix for #267 good enough.  The problem is that at the point of capitalization we don't know that the source of the string values was a dict rather than a list of raw objects.
